libs/cli/src/main/java/org/elasticsearch/cli/Terminal.java

```diff
@@ -42,6 +44,9 @@
     /** Writer to standard error - not supplied by the {@link Console} API, so we share with subclasses */
     private static final PrintWriter ERROR_WRITER = newErrorWriter();
 
+    /** We want a zero-length buffer so that we don't take too much from the input stream */
+    private static final int STDIN_READER_BUFFER_SIZE = 1;
```

Comment:
Why is this `1` if we want a zero length buffer? Do we need a bufferedreader at all if we are just reading one character at a time?

Reply:
This might not be necessary; I will take a look at getting rid of it.